### PR TITLE
Load shaders through IStorage class

### PR DESCRIPTION
**Pull request by Jupeyy**

### CODE REVIEW

**Comment by Learath2** [comment]
Fixes #955.

@Jupeyy try to put "Fixes #xxx" in your commit message next time, it automatically closes the issue when we merge.

**Comment by Learath2** [comment]
Also would be nice if you could rebase onto master

**Comment by Jupeyy** [comment]
wtf, thought that works. i used the github editor

**Comment by Jupeyy** [comment]
but yeah as @heinrich5991 pointed out its not needed at all #956

File: src/engine/client/opengl_sl.cpp [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 #include <stdio.h>
 #include <string>
 
-bool CGLSL::LoadShader(const char* pFile, int Type)
+bool CGLSL::LoadShader(IStorage* pStorage, const char* pFile, int Type)
 {
 	if (m_IsLoaded) return true;
 	IOHANDLE f;
-	f = io_open(pFile, IOFLAG_READ);
+	f = pStorage->OpenFile(pFile, IOFLAG_READ, IStorage::TYPE_ALL);
```
Comment (heinrich5991) [inline src/engine/client/opengl_sl.cpp]: The two lines above can go in one line.

**Comment by heinrich5991** [comment]
Looks good.

Can you fix the pointer style? It should be `char *pStr` in Teeworlds code, not `char* pStr`.

**Comment by Jupeyy** [comment]
@heinrich5991 was that a rule too or what ?xD
I never use that style, because the other way looks more deterministic to me, else don't we have a code style bot or something, i dunno where i placed the stars wrong

File: src/engine/client/backend_sdl.h [hunk 2]
```diff
@@ -193,6 +193,7 @@ class CCommandProcessorFragment_OpenGL3_3
 	struct SCommand_Init : public CCommandBuffer::SCommand
 	{
 		SCommand_Init() : SCommand(CMD_INIT) {}
+		class IStorage* m_pStorage;
```
Comment (heinrich5991) [inline src/engine/client/backend_sdl.h]: Star.

File: src/engine/client/backend_sdl.cpp [hunk 3]
```diff
@@ -1517,7 +1518,7 @@ void CCommandProcessor_SDL_OpenGL::RunBuffer(CCommandBuffer *pBuffer)
 
 // ------------ CGraphicsBackend_SDL_OpenGL
 
-int CGraphicsBackend_SDL_OpenGL::Init(const char *pName, int *Screen, int *pWidth, int *pHeight, int FsaaSamples, int Flags, int *pDesktopWidth, int *pDesktopHeight, int* pCurrentWidth, int* pCurrentHeight)
+int CGraphicsBackend_SDL_OpenGL::Init(const char *pName, int *Screen, int *pWidth, int *pHeight, int FsaaSamples, int Flags, int *pDesktopWidth, int *pDesktopHeight, int* pCurrentWidth, int* pCurrentHeight, IStorage* pStorage)
```
Comment (heinrich5991) [inline src/engine/client/backend_sdl.cpp]: Stars.

File: src/engine/client/backend_sdl.h [hunk 4]
```diff
@@ -305,7 +306,7 @@ class CGraphicsBackend_SDL_OpenGL : public CGraphicsBackend_Threaded
 	
 	bool m_UseOpenGL3_3;
 public:
-	virtual int Init(const char *pName, int *Screen, int *pWidth, int *pHeight, int FsaaSamples, int Flags, int *pDesktopWidth, int *pDesktopHeight, int* pCurrentWidth, int* pCurrentHeight);
+	virtual int Init(const char *pName, int *Screen, int *pWidth, int *pHeight, int FsaaSamples, int Flags, int *pDesktopWidth, int *pDesktopHeight, int* pCurrentWidth, int* pCurrentHeight, class IStorage* pStorage);
```
Comment (heinrich5991) [inline src/engine/client/backend_sdl.h]: Stars.

File: src/engine/client/graphics_threaded.h [hunk 5]
```diff
@@ -424,7 +424,7 @@ class IGraphicsBackend
 
 	virtual ~IGraphicsBackend() {}
 
-	virtual int Init(const char *pName, int *Screen, int *pWidth, int *pHeight, int FsaaSamples, int Flags, int *pDesktopWidth, int *pDesktopHeight, int* pCurrentWidth, int* pCurrentHeight) = 0;
+	virtual int Init(const char *pName, int *Screen, int *pWidth, int *pHeight, int FsaaSamples, int Flags, int *pDesktopWidth, int *pDesktopHeight, int* pCurrentWidth, int* pCurrentHeight, class IStorage* pStorage) = 0;
```
Comment (heinrich5991) [inline src/engine/client/graphics_threaded.h]: Stars.

File: src/engine/client/opengl_sl.cpp [hunk 6]
```diff
@@ -1,14 +1,15 @@
 #include "opengl_sl.h"
 #include <engine/shared/linereader.h>
+#include <engine/shared/storage.h>
 #include <vector>
 #include <stdio.h>
 #include <string>
 
-bool CGLSL::LoadShader(const char* pFile, int Type)
+bool CGLSL::LoadShader(IStorage* pStorage, const char* pFile, int Type)
```
Comment (heinrich5991) [inline src/engine/client/opengl_sl.cpp]: Stars.

File: src/engine/client/opengl_sl.h [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
 class CGLSL {
 public:
-	bool LoadShader(const char* pFile, int Type);
+	bool LoadShader(class IStorage* pStorage, const char* pFile, int Type);
```
Comment (heinrich5991) [inline src/engine/client/opengl_sl.h]: Stars. And brace not on a new line at the start of the class definition.

**Comment by Learath2** [comment]
Stars obviously belong to the right side as they are not part of the type

**Comment by heinrich5991** [comment]
Both way to write stars are completely reasonable. This is not an ideological argument, but an argument for consistency within a code base.

**Comment by Learath2** [comment]
Having the star with the variable name is reasonable in more cases, thus overall more reasonable. But yeah, no need to debate this :)

**Comment by heinrich5991** [comment]
Thanks!

**Pull request merged**